src/cdk/tree/nested-node.ts

```diff
@@ -69,10 +69,10 @@ export class CdkNestedTreeNode<T, K = T>
 
   ngAfterContentInit() {
     this._dataDiffer = this._differs.find([]).create(this._tree.trackBy);
-    if (!this._tree.treeControl.getChildren && (typeof ngDevMode === 'undefined' || ngDevMode)) {
+    if (!this._tree.treeControl?.getChildren && (typeof ngDevMode === 'undefined' || ngDevMode)) {
       throw getTreeControlFunctionsMissingError();
     }
-    const childrenNodes = this._tree.treeControl.getChildren(this.data);
+    const childrenNodes = this._tree.treeControl!.getChildren(this.data);
```

Comment:
It's not intuitive to me why on line 72 you have `treeControl?` and then you have `treeControl!`.

Reply:
Hopefully the else if makes it more understandable?